src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -1312,6 +1307,28 @@ void ZooKeeper::setServerCompletelyStarted()
         zk->setServerCompletelyStarted();
 }
 
+int32_t ZooKeeper::getConnectedHostIdx() const
+{
+    return impl->getConnectedNodeIdx();
+}
+
+String ZooKeeper::getConnectedHostPort() const
+{
+    int32_t idx = impl->getConnectedNodeIdx();
+    if (idx < 0 || static_cast<int32_t>(args.hosts.size()) <= idx)
```

Comment:
Okay, it can be -1 if we didn't find the matching address. Or after calling `finalize` (however, I don't think it's necessary to reset it in `finalize`). But how can it be `>= args.hosts.size()`? Should we throw a LOGICAL_ERROR in this case?